cpp/src/text/minhash.cu

```diff
@@ -133,13 +153,12 @@ std::unique_ptr<cudf::column> minhash(cudf::strings_column_view const& input,
                                           cudf::mask_state::UNALLOCATED,
                                           stream,
                                           mr);
-  auto d_hashes = hashes->mutable_view().data<cudf::hash_value_type>();
+  auto d_hashes = hashes->mutable_view().data<hash_value_type>();
 
-  thrust::for_each_n(
-    rmm::exec_policy(stream),
-    thrust::counting_iterator(std::size_t{0}),
-    static_cast<std::size_t>(input.size()) * static_cast<std::size_t>(cudf::detail::warp_size),
-    minhash_fn{*d_strings, seeds, width, d_hashes});
+  constexpr int block_size = 256;
+  cudf::detail::grid_1d grid{input.size() * cudf::detail::warp_size, block_size};
+  minhash_fn<HashFunction><<<grid.num_blocks, grid.num_threads_per_block, 0, stream.value()>>>(
+    *d_strings, seeds, width, d_hashes);
 
   if (seeds.size() == 1) {
```

Comment:
I'm not sure how I feel about this behavior. From the "list of seeds" overload, I expect that a list of seeds with length 1 should give back a list column of hash values with length 1 -- not a "flattened" column. The documentation mentions this is expected behavior, but flattening this on behalf of the user is not what my intuition would expect when we provide both a scalar and a list overload. If we have precedent in libcudf for doing automatic flattening (especially when scalar/list overloads exist), I'd like to know. It's fine with me to defer this to follow-up work (by filing an issue, perhaps).

Reply:
Changed this to always return lists column even if that returns only a single element per row.